ecommerce/src/main/kotlin/io/hhplus/ecommerce/coupon/application/CouponIssueWorker.kt

```
@@ -0,0 +1,112 @@
+        val userIds = couponIssueService.popPendingUsers(couponId, MAX_BATCH_SIZE)
+
+        if (userIds.isEmpty()) {
+            return
+        }
+
+        userIds.forEach { userId ->
+            try {
+                val coupon = couponDomainService.getCouponOrThrow(couponId)
+                val userCoupon = couponDomainService.issueCoupon(coupon, userId)
+                handleSuccess(couponId, userId, couponName, userCoupon)
+            } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)